homework_3/barbhaiwala/solution.js

```diff
@@ -0,0 +1,83 @@
+function NaiveCalculator(equation)
+{
+    this.splitTokens = function()
+    {
+        return equation.split(' ');
+    }
+    
+    this.applyOperation = function (operand1, operation, operand2) {
+        var resultOfOperation;
+        switch(operation)
+        {
+            case '+':
+                resultOfOperation = operand1 + operand2;
+                break;
+            case '-':
+                resultOfOperation = operand1 - operand2;
+                break;
+            case '/':
+                resultOfOperation = operand1 / operand2;
+                break;
+            case 'x':
+            case 'X':
+                resultOfOperation = operand1 * operand2;
+                break;
+            case '^':
+                resultOfOperation = Math.pow(operand1, operand2);
+                break;
+        }
+        
+        return resultOfOperation;
+    }
+    
+    this.colorize = function (param)
+    {
+        var retValue;
+        if (param > 0)
+        {
+            retValue = (param.toString() .green);
+        }
+        else if (param < 0)
+        {
+            retValue = (param.toString() .red);
+        }
+        else if (param === 0)
+        {
+            retValue = (param.toString() .yellow);
+        }
+        
+        return retValue;
+    }
+    
+    this.evaluate = function ()
+    {
+        var tokens = splitTokens();
+        var operand1;
+        var currentToken;
+        
+        for(var currentTokenCtr = 0; currentTokenCtr < tokens.length; currentTokenCtr++)
+        {
+            currentToken = tokens[currentTokenCtr]
+            var potentialOperand = parseFloat(currentToken);
```

Comment:
It's a best practice to declare all `var`'s in the same place, at the top of their scope. So, you can move this up to the declarations at the top of function and not assign anything to it yet.